Source/Core/joinUrls.js

```diff
@@ -78,6 +78,10 @@ define([
             if (baseUri.path !== '' && baseUri.path !== '/') {
                 url = url.replace(/\/?$/, '/');
                 baseUri.path = baseUri.path.replace(/^\/?/g, '');
+
+                if (baseUri.authority === '') {
```

Comment:
What exactly is this checking for?  I'd like to better understand this change before we merge.

Reply:
On line 79, a regex is used to add a slash in case a slash is missing from the end of the authority.  For example:

`http://host`  => `http://host/`

`http://host/` => `http://host/`

But with the file protocol, `authority` can be blank for local files.  So:

`file://` => `file://` -- incorrect, should be `file:///`

In this case, the regex on line 79 does not add a needed slash after an empty authority, because it sees a slash that came before the authority as satisfying the expression.

Reply:
Originally, this test was `if (url === 'file://')`, but I think checking for the blank authority is potentially more generic.

Reply:
@mramato Are you OK with this now?

Reply:
Sorry, have been out on vacation.   Can we just add a one line comment with your above explanation so future devs aren't confused.  Once that's in, I'll merge.

Reply:
Done.